RequestTypeSelector (desktop)

adamkendis opened the pull request:
Fixes #828

 - Implemented RequestTypeSelector
 - Updates redux state.filters.requestTypes on select/deselect
 - Slightly refactored redux filters.requestTypes to an array of ints (request type ids).

A few things to do:
 - Add counts to request types in TypesList. 
 - Display "All Open Requests" with a count on basic user pageload.


![typeselector](https://user-images.githubusercontent.com/40484278/105425479-52913180-5bfe-11eb-87e1-ba02bb9706a6.gif)

  - [x] Up to date with `dev` branch
  - [x] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [x] All PR Status checks are successful
  - [ ] Peer reviewed and approved

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

rufataliy commented on client/components/main/Desktop/RequestTypeSelector/SelectedTypes.jsx:
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
```
This file can be a reusable component for displaying chips. We have a lot of use cases for displaying chips. It would improve our productivity and save time.

adamkendis replied:
Yeah definitely, agree 100%. Will do that.

rufataliy commented on client/components/common/SelectorBox.jsx:
```diff
@@ -143,7 +143,7 @@ function Collapse({ children }) {
   return (
     <CollapseMUI in={expanded}>
-      <CardContent className={classes.content}>{children}</CardContent>
+      <Box className={classes.content}>{children}</Box>
```
What was the reason for changing it to Box component?

adamkendis replied:
As far as I can tell, all the `CardContent` component does is add an unwanted 24px padding on the bottom. [CardContent source code here. ](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/CardContent/CardContent.js) Replacing it with `Box` seemed better than removing the bottom padding from `CardContent` (because then what's the point of even using it) or having to add a negative bottom margin to whatever is nested in `CardContent`. 

You can see that unwanted padding in the datepicker dropdown.

![Screen Shot 2021-01-22 at 9 16 57 AM](https://user-images.githubusercontent.com/40484278/105522884-a3526a00-5c92-11eb-9008-6878363421b6.png)

rufataliy commented:
Do you think for UX improvement we could make it that if you select a selected element it will deselect it?

rufataliy commented:
Do we need to implement components separately for Desktop and Mobile? What if we will need to share something?

adamkendis commented:
> Do we need to implement components separately for Desktop and Mobile? What if we will need to share something?

Some components will be shareable between Desktop and Mobile views. I added some file structure but forgot to commit/push those. Will do that in next commit.

adamkendis commented:
@Rufat made a few changes:

1. Reusable ChipList and StyledChip components. StyledChip can be solid or outlined with the `outlined` prop. 
2. Refactored SelectedTypes to use ChipLlist/StyledChip.
3. Refactored TypesList to only display types that are currently **unselected**. If a chip is deselected in SelectedTypes, the item returns to TypesList in its original sort order instead of appending to the bottom of the list.
![Feb-04-2021 12-56-29](https://user-images.githubusercontent.com/40484278/106955313-ecbaa480-66e9-11eb-95e8-c70d7d0f52f2.gif)
